src/sinks/mod.rs

```diff
@@ -22,3 +25,77 @@ use crate::Event;
 pub type RouterSink = Box<dyn Sink<SinkItem = Event, SinkError = ()> + 'static + Send>;
 
 pub type Healthcheck = Box<dyn Future<Item = (), Error = String> + Send>;
+
+#[derive(Debug, Snafu)]
+pub enum BuildError {
```

Comment:
So I think we should avoid this large error type. Imo the best way to do this for sinks is this:

`SinkConfig`
```rust
fn build(...) -> Result<.., Box<dyn Error + Send + Sync + 'static>>;
```

```rust
#[derive(Snafu)]
enum aws_cloudwatch_logs::Error {...}

#[derive(Snafu)]
enum aws_cloudwatch_logs::HealthcheckError {...}
```

In this case, I think it is fine to repeat error variant. So what this means is that sinks/mod.rs contains no error def but exposes all errors as `Box<dyn Error>` we can then downcast later if we need the specific error type (which I doubt we will need).

Reply:
Will do.

Reply:
Done, however there were two errors that ended up repeated in a few sinks. I left them in the `sinks` module, but I could create a wrapper function or trait for them if you think that is better.